src/subtype.c

```diff
@@ -2170,7 +2170,7 @@ static jl_value_t *set_var_to_const(jl_varbinding_t *bb, jl_value_t *v JL_MAYBE_
         bb->lb = bb->ub = v;
     }
     else if (jl_is_long(v) && jl_is_long(bb->lb)) {
-        if (jl_unbox_long(v) != jl_unbox_long(bb->lb))
+        if (jl_unbox_long(v) + offset != jl_unbox_long(bb->lb))
```

Comment:
Can you expand this commit message to say a little bit more about this offset (and include the relevant test changes, if possible)?

e.g. `subtype: add tuple length offset to Vararg length when matching up typevars in intersection`